detox/android/detox/src/main/java/com/wix/detox/instruments/DetoxInstrumentsManager.java

```diff
@@ -0,0 +1,79 @@
+package com.wix.detox.instruments;
+
+import android.content.Context;
+
+import com.wix.detox.instruments.reflected.InstrumentsReflected;
+
+import java.io.File;
+
+public class DetoxInstrumentsManager {
+
+    private final Context context;
+    private final Instruments instruments;
+    private InstrumentsRecording recording;
+
+    public DetoxInstrumentsManager(Context context) {
+        this(context, getDefaultInstruments());
+    }
+
+    public DetoxInstrumentsManager(Context context, Instruments instruments) {
+        this.context = context;
+        this.instruments = instruments;
+    }
+
+    public void startRecordingAtLocalPath(String path) {
+        if (!instruments.installed()) {
+            return;
+        }
+
+        recording = instruments.startRecording(context, true, 500L, new File(path), false);
+    }
+
+    public void stopRecording() {
+        if (recording != null) {
+            recording.stop();
+            recording = null;
+        }
+    }
+
+    public void eventBeginInterval(
+            String category,
+            String name,
+            String id,
+            String additionalInfo
+    ) {
+        if (recording != null) {
+            recording.eventBeginInterval(category, name, id, additionalInfo);
+        }
+    }
+
+    public void eventEndInterval(
+            String id,
+            String eventStatus,
+            String additionalInfo
+    ) {
+        if (recording != null) {
+            recording.eventEndInterval(id, eventStatus, additionalInfo);
+        }
+    }
+
+    public void eventMark(
+            String category,
+            String name,
+            String id,
+            String eventStatus,
+            String additionalInfo
+    ) {
+        if (recording != null) {
+            recording.eventMark(category, name, id, eventStatus, additionalInfo);
+        }
+    }
+
+    public static boolean supports() {
+        return getDefaultInstruments().installed();
```

Comment:
All initializations made in order to answer this question are done static-ly, so is it really necessary to have an instance here? It seems a bit redundant and also I found it a bit confusing tbh 🤷‍♂

Reply:
Static-ly initialised only *Reflected part, which is only one of possible instruments implementations.